docs/dg/dev/third-party-module-development/develop.md

```diff
@@ -0,0 +1,159 @@
+---
+title: How to develop a Spryker module
+description: How to develop a Spryker module
+last_updated: Jun 7, 2024
+template: howto-guide-template
+
+---
+
+### Spryker architectural conventions
+
+https://docs.spryker.com/docs/dg/dev/architecture/architectural-convention.html needs to be followed in the development process.
+-  Use the samples that marked as *Module Development*.
+
+
+### How you can extend Spryker functionality from your module:
+**Option 1.** Introduce an extension point in existing Spryker module, use it. (See complete example here https://docs.spryker.com/docs/dg/dev/backend-development/plugins/plugins.html#how-to-use-a-plugin-from-another-module)
+
+**Option 2.** Adding a controller, example for project-level implementation can be found here https://docs.spryker.com/docs/dg/dev/backend-development/extend-spryker/create-modules.html#display-a-random-salutation-message, core implementation will be the same but with different namespaces.
+
+**Option 3.** Use the existing Spryker module extension point.
+
+**a.** Example of existing extension point usage.
+Were we'll extend product table with additional column **Categories** where will show the list of categories where the product is present.
+For this we'll use two extension points in the existing Spryker modules 
+- One to extend table headers with **Categories** column [ProductTableConfigurationExpanderPluginInterface](https://github.com/spryker/product-management-extension/blob/master/src/Spryker/Zed/ProductManagementExtension/Dependency/Plugin/ProductTableConfigurationExpanderPluginInterface.php)
+- And another to extend table content with corresponding column [ProductTableDataBulkExpanderPluginInterface](https://github.com/spryker/product-management-extension/blob/master/src/Spryker/Zed/ProductManagementExtension/Dependency/Plugin/ProductTableDataBulkExpanderPluginInterface.php)
+
+a. Create a module **console dev:module:create your-company-name.product-category**. (Replace the namespace and module name according to you preferences)
```

Comment:
I think, () is not needed anymore if you recommend to use the company name. What do you think?